test/mbalanceEmpty.cc

```diff
@@ -0,0 +1,55 @@
+#include <iostream>
+#include <exception>
+#include <memory>
+#include <string>
+#include <apf.h>
+#include <apfMDS.h>
+#include <apfMesh2.h>
+#include <apfPartition.h>
+#include <gmi_mesh.h>
+#include <apfMETIS.h>
+#include <lionPrint.h>
+#include <pcu_util.h>
+
+int main(int argc, char* argv[]) {
+  pcu::Init(&argc, &argv);
+  try {
+    pcu::PCU PCU;
+    if (argc != 5) {
+      if (PCU.Self() == 0)
+        std::cerr << "USAGE: <model.dmg> <mesh.smb> <inParts> <out.smd>"
+          << std::endl;
+      throw std::runtime_error("invalid arguments");
+    }
+    lion_set_verbosity(1);
+    gmi_register_mesh();
+    // load model and mesh
+    int inParts = std::stoi(argv[3]);
+    int group = PCU.Self() / inParts;
+    auto loadPCU = PCU.Split(group, 0);
+    gmi_model* model = gmi_load(argv[1]);
+    apf::Mesh2* m = nullptr;
+    if (group == 0) {
+      m = apf::loadMdsMesh(model, argv[2], loadPCU.get());
+      m->switchPCU(&PCU);
+    }
+    m = apf::expandMdsMesh(m, model, inParts, &PCU);
```

Comment:
In the added test case, this is taking the single part pipe mesh and 'expanding' it to a four part mesh which has three empty parts?

Reply:
Yes that is correct. I want to test METIS balancing with 1 or more empty parts.

I will add an assert/error message into the test case ensuring that `inParts < PCU.Peers()`.